### PR TITLE
根据配置的更具体的纹理格式加载图片，减少显存占用。

### PR DESCRIPTION
**Pull request by happyzhaole**

### CODE REVIEW

File: cocos2dx/textures/CCTextureCache.cpp [hunk 1]
```diff
@@ -229,6 +230,45 @@ CCTextureCache::~CCTextureCache()
...
+		m_pTexturesPixelFormatFile[path] = type;
+	}
+}
+
+const char* CCTextureCache::getPixelFormatStr(const char* path) {
+	std::string pathStr(path);
+	std::map<std::string, std::string>::iterator it = m_pTexturesPixelFormatDir.begin();
+	for (std::map<std::string, std::string>::iterator itEnd = m_pTexturesPixelFormatDir.end(); it != itEnd; ++it)
+	{
+		if (pathStr.find(it->first) != it->first.npos)
```
Comment (dumganhar) [inline cocos2dx/textures/CCTextureCache.cpp]: `it->first.npos` -> `std::string::npos`

File: cocos2dx/textures/CCTextureCache.cpp [hunk 2]
```diff
@@ -229,6 +230,45 @@ CCTextureCache::~CCTextureCache()
...
+		{
+			return it->second.c_str();
+		}
+	}
+
+	it = m_pTexturesPixelFormatFile.begin();
+	for (std::map<std::string, std::string>::iterator itEnd = m_pTexturesPixelFormatFile.end(); it != itEnd; ++it)
+	{
+		if (pathStr.find(it->first) != it->first.npos)
```
Comment (dumganhar) [inline cocos2dx/textures/CCTextureCache.cpp]: `it->first.npos` -> `std::string::npos`

File: cocos2dx/textures/CCTextureCache.h [hunk 3]
```diff
@@ -88,6 +91,11 @@ class CC_DLL CCTextureCache : public CCObject
      *  @js getInstance
      */
     static CCTextureCache * sharedTextureCache();
+	
+	
+	void addTexturePixelFormatDir(const char*, const char*);
+	void addTexturePixelFormatFile(const char*, const char*);
+	const char* getPixelFormatStr(const char*);
```
Comment (dumganhar) [inline cocos2dx/textures/CCTextureCache.h:98]: Missing arguments and comments

File: cocos2dx/textures/CCTextureCache.cpp [hunk 4]
```diff
@@ -859,6 +901,12 @@ void VolatileTexture::setTexParameters(CCTexture2D *t, ccTexParams *texParams)
         vt->m_texParams.wrapT = texParams->wrapT;
 }
 
+void VolatileTexture::generateMipmap(CCTexture2D *t)
+{
+	VolatileTexture *vt = findVolotileTexture(t);
+	vt->m_bHasMipmaps = true;
```
Comment (dumganhar) [inline cocos2dx/textures/CCTextureCache.cpp]: Where was `VolatileTexture::generateMipmap` invoked?

File: cocos2dx/textures/CCTextureCache.cpp [hunk 5]
```diff
@@ -412,9 +453,9 @@ CCTexture2D * CCTextureCache::addImage(const char * path)
     }
     texture = (CCTexture2D*)m_pTextures->objectForKey(pathKey.c_str());
 
+    std::string fullpath = pathKey; // (CCFileUtils::sharedFileUtils()->fullPathFromRelativePath(path));
     if (! texture) 
     {
-		std::string fullpath = pathKey; // (CCFileUtils::sharedFileUtils()->fullPathFromRelativePath(path));
```
Comment (dumganhar) [inline cocos2dx/textures/CCTextureCache.cpp:417]: Why to move this line to line `456`.

**Comment by minggo** [comment]
@happyzhaole Thanks. But we can not accept Chinese in commit message. Could you please fix it?

**Pull request closed without merging**